libcudacxx/include/cuda/std/__algorithm/copy.h

```diff
@@ -66,6 +74,34 @@ __dispatch_memmove(_Up* __result, _Tp* __first, const size_t __n)
   }
 }
 
+template <class _Tp, class _Up>
+inline _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 bool
+__constexpr_tail_overlap_fallback(_Tp* __first, _Up* __needle, _Tp* __last)
+{
+  while (__first != __last)
+  {
+    if (__first == __needle)
+    {
+      return true;
+    }
+    ++__first;
+  }
+  return false;
+}
+
+template <class _Tp, class _Up>
+inline _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 bool
+__constexpr_tail_overlap(_Tp* __first, _Up* __needle, _Tp* __last)
+{
+#if __has_builtin(__builtin_constant_p) || defined(_CCCL_COMPILER_GCC)
+  NV_IF_ELSE_TARGET(NV_IS_HOST,
+                    (return __builtin_constant_p(__first < __needle) && __first < __needle;),
+                    (return __constexpr_tail_overlap_fallback(__first, __needle, __last);))
+#else
+  return __constexpr_tail_overlap_fallback(__first, __needle, __last);
+#endif
+}
+
```

Comment:
I am not a fan of those changes. I would much rather prefer if we would write an appropriate function for bitset rather than change the copy implementation

Reply:
Eh. We already purposefully go to `memmove` and not `memcpy` in this implementation at runtime. I don't understand why we wouldn't just extend that and provide a bit of an extra guarantee that is explicitly allowed by the standard.

Reply:
@wmaxey and maybe @jrhemstad: any hot takes on this issue?

Reply:
It looks like the same hacky constexpr trickery I had to do in `<cuda/std/bit>` which makes me feel okay with what is here.

Reply:
Fine by me